vectordraw/public/js/vectordraw.js

```diff
@@ -0,0 +1,633 @@
+        if (selected.type === 'vector') {
+            this.updateVectorProperties(this.renderVector(selected.idx));
+        } else {
+            this.renderPoint(selected.idx);
+        }
+        // Enable option corresponding to selected element in menu for selecting element to edit
+        this.enableEditOption(selected);
+    };
+
+    VectorDraw.prototype.reset = function() {
+        this.pushHistory();
+        JXG.JSXGraph.freeBoard(this.board);
+        this.resetVectorProperties();
+        this.render();
+    };
+
+    VectorDraw.prototype.pushHistory = function() {
+        var state = this.getState();
+        var previous_state = _.last(this.history_stack.undo);
+        if (!_.isEqual(state, previous_state)) {
+            this.history_stack.undo.push(state);
+            this.history_stack.redo = [];
+        }
+    };
+
+    VectorDraw.prototype.undo = function() {
+        var curr_state = this.getState();
+        var undo_state = this.history_stack.undo.pop();
+        if (undo_state && !_.isEqual(undo_state, curr_state)) {
+            this.history_stack.redo.push(curr_state);
+            this.setState(undo_state);
+        }
+    };
+
+    VectorDraw.prototype.redo = function() {
+        var state = this.history_stack.redo.pop();
+        if (state) {
+            this.history_stack.undo.push(this.getState());
+            this.setState(state);
+        }
+    };
+
+    VectorDraw.prototype.getMouseCoords = function(evt) {
+        var i = evt[JXG.touchProperty] ? 0 : undefined;
+        var c_pos = this.board.getCoordsTopLeftCorner(evt, i);
+        var abs_pos = JXG.getPosition(evt, i);
+        var dx = abs_pos[0] - c_pos[0];
+        var dy = abs_pos[1] - c_pos[1];
+
+        return new JXG.Coords(JXG.COORDS_BY_SCREEN, [dx, dy], this.board);
+    };
+
+    VectorDraw.prototype.getVectorForObject = function(obj) {
+        if (obj instanceof JXG.Line) {
+            return obj;
+        }
+        if (obj instanceof JXG.Text) {
+            return this.getVectorForObject(obj.element);
+        }
+        if (obj instanceof JXG.Point) {
+            return _.find(obj.descendants, function (d) { return (d instanceof JXG.Line); });
+        }
+        return null;
+    };
+
+    VectorDraw.prototype.getVectorSettingsByName = function(name) {
+        return _.find(this.settings.vectors, function(vec) {
+            return vec.name === name;
+        });
+    };
+
+    VectorDraw.prototype.updateVectorProperties = function(vector) {
+        var vec_settings = this.getVectorSettingsByName(vector.name);
+        var x1 = vector.point1.X(),
+            y1 = vector.point1.Y(),
+            x2 = vector.point2.X(),
+            y2 = vector.point2.Y();
+        var length = vec_settings.length_factor * Math.sqrt(Math.pow(x2-x1, 2) + Math.pow(y2-y1, 2));
+        var angle = ((Math.atan2(y2-y1, x2-x1)/Math.PI*180) - vec_settings.base_angle) % 360;
+        if (angle < 0) {
+            angle += 360;
```

Comment:
Chiming in since I just happened to see this comment.

No, this can't ever be less than -360, since the previous line reduces modulo 360, so the result is in the range -359 to 359.  (If JavaScript's modulo operator were sane, the result would always be in the range 0 to 359.  However, most languages get this wrong, with Python being an exception.)